test/operator/operator_test.py

```diff
@@ -213,7 +191,9 @@ def test_linear_Op():
     x = np.random.rand(3)
     out = np.random.rand(3)
 
-    Aop = MultiplyOp(A)
+    r3 = odl.Rn(3)
+
+    Aop = MatVecOperator(r3, r3, A)
```

Comment:
While I agree on the usefulness for tests, I don't think we should change to operator interface. Why should one be forced to use a complex matrix between complex spaces when a real one would do?

IMO, in general, class constructors should not try to be too clever and guess what the user wants. They should be low-level with explicit arguments, maybe a slight level of redundancy, but most of all flexible. If a class constructor is annoying to use over time, it's easy to write a convenience function. That has the positive side effect that users see functions instead of class constructors.